addons/point_of_sale/models/pos_session.py

```diff
@@ -2255,28 +2079,29 @@ def _after_load_onboarding_data(self):
             })
 
     def load_product_frontend(self):
-        allowed = not self._pos_has_valid_product()
-        if allowed:
+        if not self._pos_has_valid_product():
             self.sudo()._load_onboarding_data()
             self._after_load_onboarding_data()
 
-        return {
-            'models_data': self.get_onboarding_data(),
-            'successful': allowed,
-        }
+        return self.get_onboarding_data()
 
     def _get_closed_orders(self):
         return self.order_ids.filtered(lambda o: o.state not in ['draft', 'cancel'])
 
     def get_pos_ui_product_pricelist_item_by_product(self, product_tmpl_ids, product_ids):
-        pricelists = self.env['product.pricelist'].search_read(**self._loader_params_product_pricelist()['search_params'])
+        params = self._load_data_params(self.config_id)
+        pricelist_params = params['search_read']['product.pricelist']
+        pricelist_item_params = params['search_read']['product.pricelist.item']
+        pricelists = self.env['product.pricelist'].search(**pricelist_params)
+
         pricelist_item_domain = [
-            ('pricelist_id', 'in', [p['id'] for p in pricelists]),
+            ('pricelist_id', 'in', [p.id for p in pricelists]),
             '|',
             '&', ('product_id', '=', False), ('product_tmpl_id', 'in', product_tmpl_ids),
             ('product_id', 'in', product_ids)]
-        return self.env['product.pricelist.item'].search_read(pricelist_item_domain, self._product_pricelist_item_fields())
 
+        pricelist_item = self.env['product.pricelist.item'].search_read(pricelist_item_domain, pricelist_item_params['fields'], load=False)
+        return {'product.pricelist.item': pricelist_item}
```

Comment:
I agree with the pattern that we should return the result of `search_read` so that the result is easily compatible to the objects in the client, however, I don't agree with labelling them like this. The caller should know the model of the object that is returned to it. So, in my opinion, we should just return the records as array.

```suggestion
        return pricelist_items
```

Reply:
Ok I will change that